### ISSUE TITLE
[Bug]: issue with the version number (I think)

### ISSUE THREAD

**Issue by TomDonoghue** [NeurodataWithoutBorders/pynwb#1522]
### What happened?

There appears to be some kind of quirk going on with the version number that `pynwb` gets when installing a local copy. 

Long story short (details below), I installed a local copy of current dev branch, and get the following version number: 2.0.1.post.dev36

This version number seems like it must be wrong, in that the main release on PYPI is 2.1.0, and presumably the current dev version should be ahead of this. I have tried to look into this a bit, but not being familiar with versioneer and the approach used here, I'm having a hard time telling how version gets defined (though I could look into this a bit more if someone can provide some pointers). 

I double checked the version number from pynwb when installing from PYPI, and that works (shows 2.1.0). 

### Steps to Reproduce

I cloned and locally installed 
[my fork](https://github.com/TomDonoghue/pynwb),
and then run:

```python
from pynwb import __version__
__version__
```
Which prints: '2.0.1.post.dev36'

I have double checked that my fork is up to date from source. 
I am using the `dev` branch, which as far as I can tell is equivalent to `main`.
```


### Traceback

_No response_

### Operating System

macOS

### Python Executable

Conda

### Python Version

3.8

### Package Versions

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/.github/CODE_OF_CONDUCT.rst)
- [X] Have you checked the [Contributing](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/docs/CONTRIBUTING.rst) document?
- [X] Have you ensured this bug was not already [reported](https://github.com/NeurodataWithoutBorders/pynwb/issues)?

**Comment by rly** [comment]
In your fork, can you run `git tag -l` and paste it here? If 2.1.0 is not the third from the bottom, then please try `git fetch --tags upstream` (replace upstream with the name of the remote for the source repo, or replace it with the git URL of the source repo).

**Comment by TomDonoghue** [comment]
When I first ran `git tag -l`, the last few entries I saw are:
```
2.0.0
2.0.1
latest
latest-tmp-tmp
```

After running `git fetch --tags https://github.com/NeurodataWithoutBorders/pynwb`, then it was updated and the third from the last became `2.1.0`, as it seems it should be. 

So I guess my fork was somehow lagging behind on the tags. I don't remember exactly when I forked this - it might have been before the newest release, which I guess would explain this. In that case perhaps this is all working as intended - with the caveat that any fork that was created before a tagged release will get an older version number, even of the commit log is up to date?
